services/datamanager/datasync/sync.go

```diff
@@ -81,23 +82,43 @@ func NewManager(identity string, client v1.DataSyncServiceClient, logger logging
 	cancelCtx, cancelFunc := context.WithCancel(context.Background())
 	logger.Debugf("Making new syncer with %d max threads", maxSyncThreads)
 	ret := syncer{
-		partID:             identity,
-		client:             client,
-		logger:             logger,
-		cancelCtx:          cancelCtx,
-		cancelFunc:         cancelFunc,
-		arbitraryFileTags:  []string{},
-		inProgress:         make(map[string]bool),
-		syncErrs:           make(chan error, 10),
-		syncRoutineTracker: make(chan struct{}, maxSyncThreads),
-		captureDir:         captureDir,
+		partID:            identity,
+		client:            client,
+		logger:            logger,
+		cancelCtx:         cancelCtx,
+		cancelFunc:        cancelFunc,
+		arbitraryFileTags: []string{},
+		inProgress:        make(map[string]bool),
+		syncErrs:          make(chan error, 10),
+		filesToSync:       make(chan string), //shold this be buffered?
+		// syncRoutineTracker: make(chan struct{}, maxSyncThreads),
+		captureDir: captureDir,
 	}
 	ret.logRoutine.Add(1)
 	goutils.PanicCapturingGo(func() {
 		defer ret.logRoutine.Done()
 		ret.logSyncErrs()
 	})
 
+	for i := 0; i < maxSyncThreads; i++ {
+		ret.backgroundWorkers.Add(1)
+		go func() {
+			defer ret.backgroundWorkers.Done()
+			defer fmt.Print("exiting from sync thread, should not see")
```

Comment:
I assume the intention is to take this out once we're confident things are working correctly.

Reply:
yep!